src/osp/Active/ActiveScene.cpp

```diff
@@ -39,196 +40,41 @@ void ACompCamera::calculate_projection()
-}
-
-void ActiveScene::hier_cut(ActiveEnt ent)
-{
-    auto &entHier = m_registry.get<ACompHierarchy>(ent);
-
-    // TODO: deal with m_depth
-
-    // Set sibling's sibling's to each other
-
-    if (m_registry.valid(entHier.m_siblingNext))
-    {
-        m_registry.get<ACompHierarchy>(entHier.m_siblingNext).m_siblingPrev
-                = entHier.m_siblingPrev;
-    }
-
-    if (m_registry.valid(entHier.m_siblingPrev))
-    {
-        m_registry.get<ACompHierarchy>(entHier.m_siblingPrev).m_siblingNext
-                = entHier.m_siblingNext;
-    }
-
-    // deal with parent
-
-    auto &parentHier = m_registry.get<ACompHierarchy>(entHier.m_parent);
-    parentHier.m_childCount --;
-
-    if (parentHier.m_childFirst == ent)
-    {
-        parentHier.m_childFirst = entHier.m_siblingNext;
-    }
-
-    entHier.m_parent = entHier.m_siblingNext = entHier.m_siblingPrev
-            = entt::null;
-}
-
-void ActiveScene::on_hierarchy_construct(ActiveReg_t& reg, ActiveEnt ent)
-{
-    m_hierarchyDirty = true;
-}
-
-void ActiveScene::on_hierarchy_destruct(ActiveReg_t& reg, ActiveEnt ent)
-{
-    m_hierarchyDirty = true;
-}
-
 void ActiveScene::update()
 {
-
     m_updateOrder.call(*this);
-}
-
-
-void ActiveScene::update_hierarchy_transforms()
-{
-    if (m_hierarchyDirty)
-    {
-        // Sort by level, so that objects at the top of the hierarchy are
-        // updated first
-        m_registry.sort<ACompHierarchy>([](ACompHierarchy const& lhs,
-                                           ACompHierarchy const& rhs)
-        {
-            return lhs.m_level < rhs.m_level;
-        }, entt::insertion_sort());
-
-        m_registry.sort<ACompTransform, ACompHierarchy>();
-
-        m_hierarchyDirty = false;
-    }
 
-    auto view = m_registry.view<ACompHierarchy, ACompTransform>();
-
-    for(ActiveEnt entity : view)
-    {
-        auto& hierarchy = view.get<ACompHierarchy>(entity);
-        auto& transform = view.get<ACompTransform>(entity);
-
-        if (hierarchy.m_parent == m_root)
-        {
-            // top level object, parent is root
-
-            transform.m_transformWorld = transform.m_transform;
-
-        }
-        else
-        {
-            auto& parentTransform
-                    = view.get<ACompTransform>(hierarchy.m_parent);
-
-            // set transform relative to parent
-            transform.m_transformWorld = parentTransform.m_transformWorld
-                                          * transform.m_transform;
-        }
-    }
+    // TODO: FunctionOrder is way too inconvenient to deal with, so for now the
+    //       delete systems are called here
+    SysHierarchy::update_delete(*this);
+    update_delete(*this);
 }
 
 void ActiveScene::draw(ActiveEnt camera)
 {
+    SysHierarchy::sort(*this);
+    SysRender::update_hierarchy_transforms(*this);
+
     auto renderView = m_registry.view<ACompRenderingAgent, ACompPerspective3DView, ACompRenderer>();
     for (auto [e, agent, view, renderer] : renderView.each())

```

Comment:
small preference to change "e" to something more intuitive.

Reply:
Could change it to `ent` I guess but this arrangement is used pretty much everywhere and if you're familiar with EnTT it's obvious since the first element of the `each()` closure is always the entity

Reply:
New contributors probably won't realize that right away.

Better to err on the side of welcoming new devs when it's low cost to do.

Reply:
Going to make this `const&` too